hnn_core/opt_toy_example/metrics.py

```diff
@@ -1,89 +1,65 @@
-from hnn_core import simulate_dipole
 import numpy as np
-from mne.time_frequency import psd_array_multitaper
 
+from hnn_core import simulate_dipole, MPIBackend
 
-def _rmse_evoked(net, param_names, target_statistic, predicted_params,
-                 f_bands, weights, _set_params):
-    """The objective function for evoked responses.
+from scipy.signal import resample
 
-       Parameters
-       -----------
-       net : Network
-       param_names : dictionary
-           Parameters to change.
-       target_statistic : ndarray
-           The recorded dipole.
-       predicted_params : list
-           Parameters selected by the optimizer.
 
-       Returns
-       -------
-       rmse : normalized RMSE between recorded and simulated dipole
+def _rmse_evoked(net, initial_params, set_params, predicted_params,
+                 update_params, obj_values, scale_factor, smooth_window_len,
+                 tstop, target_statistic):
     """
-
-    # get network with predicted params
-    new_net = _set_params(net, param_names, predicted_params)
-    # simulate dipole
-    dpl = simulate_dipole(new_net, tstop=500, n_trials=1)[0]
-
-    # smooth & scale (scale must be passed in by user)
-    dpl.scale(1000)
-    dpl.smooth(20)
-
-    # calculate rmse
-    obj = np.sqrt(((dpl.data['agg'] - target_statistic)**2).sum()
-                   / len(dpl.times)) / (max(target_statistic)
-                                        - min(target_statistic))
-    return obj
-
-
-def _rmse_rhythmic(net, param_names, target_statistic, predicted_params,
-                   f_bands, weights, _set_params):
-    """The objective function for evoked responses.
-
-       Parameters
-       -----------
-       net : Network
-       param_names : dictionary
-           Parameters to change.
-       target_statistic : ndarray
-           Recorded dipole.
-       predicted_params : list
-           Parameters selected by the optimizer.
-       compute_psd :
-           ...
-
-       Returns
-       -------
-       rmse : normalized RMSE between recorded and simulated dipole
+    The objective function for evoked responses.
+
+    Parameters
+    ----------
+    net : Network
+        The network object.
+    initial_params : dict
+        Keys are parameter names, values are initial parameters.
+    set_params : func
+        User-defined function that sets network drives and parameters.
+    predicted_params : list
+        Parameters selected by the optimizer.
+    update_params : func
+        Function to update param_dict.
+    scale_factor : float
+        The dipole scale factor.
+    smooth_window_len : float
+        The smooth window length.
+    tstop : float
+        The simulated dipole's duration.
+    target_statistic : ndarray
```

Comment:
```suggestion
    target : ndarray
```
?
Since it's really just a signal rather than a "statistic" being calculated on top of it

Reply:
Feel free to disagree ;), `target` is pretty vague